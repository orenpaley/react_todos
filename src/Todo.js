// Renders a single Todo Element
// msg - text of todo, id = id of todo, complete marks todo as complete
// remove function to remove todo, isComplete is a boolean for if todo is completed

const Todo = ({ msg, id, complete, remove, isComplete }) => {
  // delete a todo
  function handleRemove(e) {
    e.preventDefault();
    remove(e.target.parentElement);
    // complete or cancel completion of todo
  }
  function handleComplete(e) {
    e.preventDefault();
    complete(e.target.parentElement);
  }
  function handleEdit(e) {}
  // checks if todo is completed or not then renders todo appropriately
  if (!isComplete) {
    return (
      <div className="Todo" id={id}>
        <h4
          style={{ color: "orange", textDecoration: "none" }}
          onClick={handleComplete}
        >
          {msg}
        </h4>
        <button onClick={handleRemove}>X</button>
      </div>
    );
  } else {
    return (
      <div className="Todo" id={id}>
        <h4
          style={{ color: "green", textDecoration: "line-through" }}
          onClick={handleComplete}
        >
          {msg}
        </h4>
        <button onClick={handleRemove}>X</button>
      </div>
    );
  }
};
export default Todo;
