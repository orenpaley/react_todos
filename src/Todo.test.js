import Todo from "./Todo";

import { render } from "@testing-library/react";

// test("renders learn react link", () => {
//   render(<BoxList />);
//   const linkElement = screen.getByText(/learn react/i);
//   expect(linkElement).toBeInTheDocument();
// });

it("renders without crashing", () => {
  render(<Todo id={1} key={1} msg={"TodoTestMsg"} isComplete={false} />);
});
it("matches snapshot", function () {
  const { asFragment } = render(
    <Todo id={1} key={1} msg={"TodoTestMsg"} isComplete={false} />
  );
  expect(asFragment()).toMatchSnapshot();
});
