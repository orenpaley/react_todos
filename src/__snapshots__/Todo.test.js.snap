// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Todo"
    id="1"
  >
    <h4
      style="color: orange; text-decoration: none;"
    >
      TodoTestMsg
    </h4>
    <button>
      X
    </button>
  </div>
</DocumentFragment>
`;
